q6.c

```diff
@@ -0,0 +1,122 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <pthread.h>
+#define tam 10
+
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+pthread_mutex_t db = PTHREAD_MUTEX_INITIALIZER;
+
+
+int *regiaocritica;
```

Comment:
Se tua região crítica tem tamanho fixo, não precisaria ser alocada
```suggestion
int regiaocritica[TAM];
```